subxt/src/backend/mod.rs

```diff
@@ -325,9 +325,416 @@ pub enum TransactionStatus<Hash> {
 
 /// A response from calls like [`Backend::storage_fetch_values`] or
 /// [`Backend::storage_fetch_descendant_values`].
+#[cfg_attr(test, derive(serde::Serialize, Clone, PartialEq, Debug))]
 pub struct StorageResponse {
     /// The key.
     pub key: Vec<u8>,
     /// The associated value.
     pub value: Vec<u8>,
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    mod legacy {
+        use std::{
+            collections::{HashMap, VecDeque},
+            sync::Arc,
+        };
+        use tokio::sync::{mpsc, Mutex};
+
+        use super::rpc::{RpcClient, RpcClientT};
+        use crate::backend::rpc::RawRpcSubscription;
+        use crate::backend::BackendExt;
+        use crate::{
+            backend::{
+                legacy::rpc_methods::Bytes, legacy::rpc_methods::RuntimeVersion,
+                legacy::LegacyBackend, StorageResponse,
+            },
+            error::RpcError,
+        };
+        use futures::StreamExt;
+        use serde::Serialize;
+        use serde_json::value::RawValue;
+        use subxt_core::{config::DefaultExtrinsicParams, Config};
+        type RpcResult<T> = Result<T, RpcError>;
+        type Item = RpcResult<String>;
+
+        struct MockDataTable {
+            items: HashMap<Vec<u8>, VecDeque<Item>>,
+        }
+
+        impl MockDataTable {
+            fn new() -> Self {
+                MockDataTable {
+                    items: HashMap::new(),
+                }
+            }
+
+            fn from_vec<'a, T: Serialize, I: IntoIterator<Item = (&'a str, RpcResult<T>)>>(
```

Comment:
from_iter is probably more accurate but doesn't matter that much

Reply:
That'd be more consistent yup :)